cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -91,12 +91,24 @@ const int32_t* getFirstColumn(const facebook::velox::RowVector& rv) {
   VELOX_CHECK(rv.childrenSize() > 0, "RowVector missing partition id column.");
 
   auto& firstChild = rv.childAt(0);
-  VELOX_CHECK(firstChild->type()->isInteger(), "RecordBatch field 0 should be integer");
+  VELOX_CHECK(firstChild->isFlatEncoding(), "Partition id (field 0) is not flat encoding.");
+  VELOX_CHECK(
+      firstChild->type()->isInteger(),
+      "Partition id (field 0) should be integer, but got {}",
+      firstChild->type()->toString());
 
   // first column is partition key hash value or pid
   return firstChild->asFlatVector<int32_t>()->rawValues();
 }
 
+facebook::velox::VectorPtr flatChildAt(const facebook::velox::RowVector& rv, facebook::velox::column_index_t idx) {
+  auto column = rv.childAt(idx);
+  if (column->isLazy()) {
```

Comment:
What if column is complex datatype? Could this situation happens? The `struct` column is not lazy and its child column is lazy.

Reply:
The complex types will be handled by `PrestoSerializer` properly.